src/Domain/DoctrineDonationRepository.php

```diff
@@ -90,10 +87,28 @@ public function allocateMatchFunds(Donation $donation): string
             $amountNewlyMatched = bcadd($amountNewlyMatched, $newWithdrawalAmount, 2);
         }
 
-        $this->getEntityManager()->flush(); // Flush `$newWithdrawals` if any.
+        try {
+            // Flush `$newWithdrawals` if any and updates to DB copies of CampaignFundings.
+            $this->getEntityManager()->flush();
+        } catch (\Throwable $exception) {
+            $this->getLogger()->error(sprintf(
+                'Failed to flush DB for new withdrawals so donation UUID %s did not get matching: %s',
+                $donation->getUuid()->toString(),
+                $exception->getMessage(),
+            ));
+
+            // todo pick one
+            $this->matchingAdapter->releaseNewlyAllocatedFunds();
+            //$this->reverseMatchingAllocations($donation, $newWithdrawals, $likelyAvailableFunds);
```

Comment:
noel thinks probably pick `releaseNewlyAllocatedFunds` (I haven't looked enough yet to have my own view)